### PR TITLE
[MOD-4830] Fixing numric range cursor

### PR DESCRIPTION
**Pull request by ephraimfeldblum**
[MOD-4830](https://redislabs.atlassian.net/browse/MOD-4830)
Fixed formatting of fieldname so that it will be found correctly. Prevented cursor from immediately aborting iteration.
Also handle `IndexReader_Reopen` with Union Iterator.

[MOD-4830]: https://redislabs.atlassian.net/browse/MOD-4830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/pytests/test_cursors.py [hunk 1]
```diff
@@ -141,3 +141,23 @@ def testLeaked(env):
     # Test ensures in CursorList_Destroy() checks shutdown with remaining cursors
     loadDocs(env)
     env.expect('FT.AGGREGATE idx * LOAD 1 @f1 WITHCURSOR COUNT 1 MAXIDLE 1')
+
+def testNumericCursor(env):
+    # conn = getConnectionByEnv(env)
+    idx = 'foo'
+    field = 'ff'
+    env.expect('FT.CREATE', idx, 'ON', 'HASH', 'prefix', 1, idx, 'SCHEMA', field, 'NUMERIC').ok()
```
Comment (oshadmi) [inline tests/pytests/test_cursors.py:149]: Consider removing the PREFIX to make this a minimal reproducible example.

File: src/numeric_index.c [hunk 2]
```diff
@@ -856,7 +856,8 @@ void NumericRangeIterator_OnReopen(void *privdata) {
   IndexIterator *it = nu->it;
   IndexReader *ir = it->ctx;
 
-  RedisModuleString *numField = RedisModule_CreateString(NULL, nu->fieldName, strlen(nu->fieldName));
+  RedisSearchCtx sctx = (RedisSearchCtx)SEARCH_CTX_STATIC(RSDummyContext, sp);
+  RedisModuleString *numField = fmtRedisNumericIndexKey(&sctx, nu->fieldName);
```
Comment (oshadmi) [inline src/numeric_index.c:860]: Can use `IndexSpec_GetFormattedKey` which caches formatted key names

**Review by oshadmi — COMMENTED** [review]
Looks good!
Few comments

File: tests/pytests/test_cursors.py [hunk 3]
```diff
@@ -141,3 +141,23 @@ def testLeaked(env):
...
     loadDocs(env)
     env.expect('FT.AGGREGATE idx * LOAD 1 @f1 WITHCURSOR COUNT 1 MAXIDLE 1')
+
+def testNumericCursor(env):
+    conn = getConnectionByEnv(env)
+    idx = 'foo'
+    field = 'ff'
+    env.expect('FT.CREATE', idx, 'ON', 'HASH', 'SCHEMA', field, 'NUMERIC').ok()
+    for x in range(1, 2):
+        conn.execute_command('HSET', '{}_{}'.format(idx, x), field, x)
```
Comment (rafie) [inline tests/pytests/test_cursors.py:151]: `f"{idx}_{x}"` is better

File: src/numeric_index.c [hunk 4]
```diff
@@ -856,11 +856,10 @@ void NumericRangeIterator_OnReopen(void *privdata) {
   IndexIterator *it = nu->it;
   IndexReader *ir = it->ctx;
 
-  RedisModuleString *numField = RedisModule_CreateString(NULL, nu->fieldName, strlen(nu->fieldName));
+  RedisModuleString *numField = IndexSpec_GetFormattedKeyByName(sp, nu->fieldName, INDEXFLD_T_NUMERIC);
   NumericRangeTree *rt = openNumericKeysDict(sp, numField, 0);
-  RedisModule_FreeString(NULL, numField);
   
-  if (!rt || rt->revisionId != nu->lastRevId) {
+  if (!rt || rt->revisionId != nu->lastRevId /* || ir->gcMarker != ir->idx->gcMarker */) {
```
Comment (oshadmi) [inline src/numeric_index.c:862]: **Temp** fix just for releasing a milestone
```suggestion
  if (!rt || rt->revisionId != nu->lastRevId || ir->gcMarker != ir->idx->gcMarker) {
```

**Review by oshadmi — DISMISSED** [review]

**Review by oshadmi — APPROVED** [review]

**Pull request merged**